spec/helpers/settings_helper_spec.rb

```diff
@@ -0,0 +1,197 @@
+#-- encoding: UTF-8
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+require 'spec_helper'
+require 'ostruct'
+
+describe SettingsHelper, type: :helper do
+  include Capybara::RSpecMatchers
```

Comment:
@ulferts would like to get your feedback on using `Capybara::RSpecMatchers` here. `Capybara::RSpecMatchers` are often useful outside of integration specs (indeed we also use them in the API v3 request specs), but I'm concerned they might be the wrong level of abstraction here…

In integration specs we're typically interested in the data being displayed and not the exact structure of the DOM. In these Helper specs, we _are interested_ in the DOM structure. If Rails or our own application code happens to append additional attributes, that is also something we care about.

However, I find doing exact String matching for Helper output to be cumbersome (you need to ignore whitespace differences, for a start). Changes in attribute order (that have no effect on what is actually rendered in the browser) mean a lot of [extra spec maintenance](https://github.com/opf/openproject/commit/4b90766bc0c00d15c7c1841897c768189ade7a7f).

Alternatives are Rails' `assert_select`, but I don't really care for the nested blocks API.

WDYT?